### PR TITLE
Fix load-factory redirection in case of policies.create='peruser'

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
This PR fixed load-factory redirections for the cases with policies.create="peruser".

### What issues does this PR fix or reference?
Needed for https://github.com/eclipse/che/issues/14554

### CODE REVIEW

**Comment by olexii4** [comment]
This PR also included a fix for [#14552](https://github.com/eclipse/che/issues/14552)
because it was needed for manual testing.
Screenshot from 2019-10-10 09-29-49
Screenshot from 2019-10-10 09-30-33

**Review by musienko-maxim — APPROVED** [review]

**Review by AndrienkoAleksandr — APPROVED** [review]

File: dashboard/src/app/factories/load-factory/load-factory.controller.ts [hunk 1]
```diff
@@ -244,7 +247,32 @@ export class LoadFactoryController {
...
+            if (angular.isDefined(factoryUrl)) {
+              const factoryUrlObj = new (window as any).URL(factoryUrl);
+              const pathIsCorrect = `${factoryUrlObj.origin}${factoryUrlObj.pathname}` === this.routeParams.url;
+              if (pathIsCorrect === false) {
+                return false;
+              }
+
+              let factoryUrlParamsNumber = 0;
+              let hasExtraKey = false;
+              factoryUrlObj.searchParams.forEach((volume: string, key: string) => {
```
Comment (mmorhun) [inline dashboard/src/app/factories/load-factory/load-factory.controller.ts]: Up to you, but seems  `for .. of` loop is clearer here (and you may return just in place)

**Review by akurinnoy — APPROVED** [review]

**Review by mmorhun — APPROVED** [review]

File: dashboard/src/app/factories/load-factory/load-factory.controller.ts (hunk 1 above)
Comment (olexii4) [reply dashboard/src/app/factories/load-factory/load-factory.controller.ts]: ok

File: dashboard/src/app/factories/load-factory/load-factory.controller.ts [hunk 2]
```diff
@@ -244,7 +247,32 @@ export class LoadFactoryController {
           if (this.factory.id) {
             return this.factory.id === (w.attributes as any).factoryId;
           } else if (this.routeParams.url){
-            return this.routeParams.url === (w.attributes as any).factoryurl;
+            const factoryUrl = (w.attributes as any).factoryurl;
+            // compare factory URL and route params
+            if (angular.isDefined(factoryUrl)) {
+              const factoryUrlObj = new (window as any).URL(factoryUrl);
+              const pathIsCorrect = `${factoryUrlObj.origin}${factoryUrlObj.pathname}` === this.routeParams.url;
+              if (pathIsCorrect === false) {
```
[the remark this answers is not in the document]
Comment (olexii4) [reply dashboard/src/app/factories/load-factory/load-factory.controller.ts]: done

**Review by evidolob — APPROVED** [review]

**Comment by Ohrimenko1988** [comment]
ci-build

**Comment by Ohrimenko1988** [comment]
ci-build

**Pull request merged**